Agent installation docs (#135)
jetstack/preflight: README.md

**[Preflight _agent_](#agent).**
While this is happening the docs may be inconsistent and the repo structure may
change.

> [Docs for this component can still be found here](./docs/installation_manual_in_cluster.md).

Reviewer: ```suggestion
[Docs for the check tool can still be found here](./docs/installation_manual_in_cluster.md).
```

Just to clarify as we mentioned two components in the previous sentence.

Author: fixed in aff5c9841c73a6736816db01edf4552ed1edf1df

> [Docs for this component can still be found here](./docs/installation_manual_in_cluster.md).

Reviewer: ```suggestion
[Docs for this component can still be found here](./docs/check.md).
```

Author: fixed in aff5c9841c73a6736816db01edf4552ed1edf1df

![Preflight package structure diagram](./docs/images/preflight_package.png)

## Installation

This section outlines how to install Preflight.

### Agent

The following instructions walk through the installation of the Preflight agent
to gather data about cluster pods and send them to the backend for analysis.

**To complete the secret manifest below, you will need to have a Preflight
token.**

> First create a namespace for the preflight components:

Reviewer: I feel like these manifests should be in files in `examples/` or `deploy/` so they are easier for user to consume, though this might be an issue for a future PR.

Author: I think my other comment touches on this.

metadata:
  name: preflight-agent-cluster-viewer
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  # will be able to view all resources, but not rbac and secrets
  name: view
subjects:
- kind: ServiceAccount
  name: agent
  namespace: preflight
```

Finally deploy the agent:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: agent
  namespace: preflight
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: agent
  template:
    metadata:
      labels:
        app.kubernetes.io/name: agent
    spec:
      serviceAccountName: agent
      volumes:
      - name: config
        secret:
          secretName: agent-config
      containers:
      - name: agent
        image: quay.io/jetstack/preflight:7d4fa467258b7592d68fd660f1fd1d42e7332231
        args:
        - "agent"
        - "-c"
        - "/etc/secrets/preflight/agent/config.yaml"
        volumeMounts:
        - name: config
          mountPath: "/etc/secrets/preflight/agent"
          readOnly: true
        resources:
          requests:
            memory: "200Mi"
            cpu: "200m"
          limits:
            memory: "200Mi"
            cpu: "200m"
```

> ### Preflight V1 (check)

Reviewer: Maybe to make things cleaner all the info about check should be moved out to another file in `docs/` and just mentioned and linked once above. Might also be best for another PR.

Author: I can move them out if that's the popular vote.

My reasoning here was that we have documentation in a lot of places - this is becoming harder to keep in sync as we deprecate check. 

Until then, I wanted to keep the changes minimal and quite self contained until we're ready to remove check references completely...

Reviewer: Yeah I can see the reasoning there. I think I just don't like the cluttered feel in the mean time but I can always just get over it.

My thinking was that we actually move all check specific stuff into `docs/check.md`, refer to this once in `README.md` and otherwise everything can focus on the agent and new SaaS way of working.

Author: Sounds reasonable, I'll see what that looks like.

Author: ffe6c83351954fec31df77d8c638f3584b4fbcb7 is an attempt at this.

Author: I've tried to move all the check docs and examples to one place - leaving the new readme as a clear process for installing the agent.
